codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageDecoder;
+import io.netty.util.ByteProcessor;
+import io.netty.util.CharsetUtil;
+import io.netty.util.internal.PlatformDependent;
+
+import java.util.List;
+
+/**
+ * Decodes the Redis protocol into {@link RedisMessage} objects following
+ * <a href="http://redis.io/topics/protocol">RESP (REdis Serialization Protocol)</a>.
+ *
+ * {@link RedisMessage} parts can be aggregated to {@link RedisMessage} using
+ * {@link RedisArrayAggregator} or processed directly.
+ */
+public class RedisDecoder extends ByteToMessageDecoder {
+
+    private static final int CRLF_LENGTH = 2;
+    private static final int NULL_RESPONSE_LENGTH = -1;
+    private static final int REDIS_MESSAGE_MAX_LENGTH = 512 * 1024 * 1024; // 512MB
```

Comment:
this is also used by the `RedisBulkStringAggregator`. Can we share this define in a common place?